### ISSUE TITLE
Add moduleId and chunkId Hashing.

### ISSUE THREAD

**Issue by maxaon** [wix/yoshi#484]
# 🙋 Feature Request
Add **HashedModuleIds** and **HashedChunksIds** plugins

### 😯 Current Behavior
Currently wepalck generates numeric ids for chunks and modules. If page has more then one module with dynamic imports it leads to module and chunk overwrite. 

### 🤔 Expected Behavior

Chunks and Modules and etc should have unique ids to prevent id collisions. 

### 💁 Possible Solution 

Add [HashedModuleIdsPlugin](https://webpack.js.org/plugins/hashed-module-ids-plugin/) and [HashedChunkidsPlugin](https://github.com/hxlniada/webpack-hashed-chunkids) to webpack plugin list.
```js
new webpack.HashedModuleIdsPlugin({
  hashFunction: 'md5',
  hashDigest: 'hex',
  hashDigestLength: 20
}),
new (require('webpack-hashed-chunkids'))({
  hashFunction: 'md5',
  hashDigest: 'hex',
  hashDigestLength: 20
}),
```

### 🔦 Context 

We have issue when Art Store (2.573.0) and Music Manager (1.69.0) are installed in one site. 
Issue flow:

1. MusicManager module is loaded. It asynchronously importing chunk with id 0; 'webpackJsonp' is installed. 

2. Art Stores Module is executed and 'webpackJsonp' is overwritten. 

3. Chunk 0 is loaded. webpackJsonp is called and all modules from chunk are written to internal cache of Art Stores Module. webpackJsonp from MusicManager is called and chunk successfully working.

4. Art stores executing some code and requiring some module. Require function return module from chunk of MusicManager.

3. Errors happens. In our case: 
```
Uncaught (in promise) TypeError: Cannot read property 'call' of undefined
    at __webpack_require__ (art-store-module.bundle.js:52)
    at Object.__extends (0.chunk.js:43002)
    at __webpack_require__ (art-store-module.bundle.js:52)
    at Object.defineProperty.value (0.chunk.js:10709)
    at __webpack_require__ (art-store-module.bundle.js:52)
    at Object.exports.isFunction (0.chunk.js:4871)
    at __webpack_require__ (art-store-module.bundle.js:52)
    at dispatchXhrRequest (art-store-module.bundle.js:4462)
    at new Promise (<anonymous>)
    at xhrAdapter (art-store-module.bundle.js:4368)
    at dispatchRequest (art-store-module.bundle.js:7814)
```

**Comment by ranyitz** [comment]
@maxaon Thanks for opening this issue and for the research.

Adding a hash for each module and for each chunk will increase bundle sizes across projects in Wix, we would like to avoid that as much as we can.

The problem introduced here indicates that 2 bundles with dynamic import cannot live on the same page using the current setup because they share the same `jsonpFunction` on the global scope (window).

#497 should change only the `jsonpFunction` according to the project name, and by that enable full isolation without changing the inner chunk/module names.

**Comment by ronami** [comment]
Solved in #497 and released in v3.4.4. @maxaon, please confirm the problem is now solved.
